gateway/mw_oas_validate_request.go

```diff
@@ -52,7 +52,13 @@ func (k *ValidateRequest) ProcessRequest(w http.ResponseWriter, r *http.Request,
 		return nil, http.StatusOK
 	}
 
-	router, err := gorillamux.NewRouter(&k.Spec.OAS.T)
+	// replacing servers object to just have listen path so that router.FindRoute(r) will not fail with strict hostname check
+	oasSpec := k.Spec.OAS.T
+	oasSpec.Servers = openapi3.Servers{
+		{URL: k.Spec.Proxy.ListenPath},
```

Comment:
Do we really need URL line here. Can be it just emty servers object?
I can envision issues when ListenPath contains regexp.

Reply:
adding empty servers won't work since we're proxying behind a listen path. 
kin/openapi3 internally use gorilla/mux router matching. 
When OAS servers object is empty, the routes registered will be what's in the OAS doc. eg `/pet`, `/store/order`. 
However, when the listen path is say `petstore` and not `/`, the request url will be `/petstore/pet`, `/petstore/store/order` etc. 
This won't match with the registered routes from OAS spec.

Reply:
I see. Can you try to add a test with listen path containing regexp? (we have examples of it in our tests).

Reply:
added tests with listen path having regexp